src/adt/test_utils.rs

```diff
@@ -137,77 +137,92 @@ pub async fn test_wrong_guard<const WORD_LENGTH: usize, Memory>(
 /// Spawns multiple threads to perform concurrent counter increments.
 /// Uses retries to handle write contention between threads.
 /// Verifies the final counter matches the total number of threads.
+///
+///
+/// # Arguments
+///
+/// * `memory` - The Memory ADT implementation to test.
+/// * `seed` - The seed used to initialize the random number generator.
+/// * `n_threads` - The number of threads to spawn. If None, defaults to 100.
 pub async fn test_guarded_write_concurrent<const WORD_LENGTH: usize, Memory>(
     memory: &Memory,
     seed: [u8; KEY_LENGTH],
+    n_threads: Option<usize>,
```

Comment:
these are not threads anymore:
```suggestion
    n_workers: Option<usize>,
```